ide-support/revdocsparser.livecodescript

```diff
@@ -947,13 +947,16 @@ private function __elementRegex
    return "^ *([tT]he [rR]esult|\w+) *(?:\(( *(?:optional |in |out |inout )?<?\w*>?) *\))? *: *(.*)"
 end __elementRegex
 
-/*
-Utility function to extract any elements matching the docs element style
-*/
-function revDocsExtractElements pText
-   # Get the element-matching regex
+private function __arrayElementRegex
+   # Regex used to extract sub-array elements
+   # Essentially the same as the element regex except quotation marks are allowed
+   # as keys can obviously be string literals.
+   return "^ *([\w" & quote & "]+) *(?:\(( *(?:optional |in |out |inout )?<?\w*>?) *\))? *: *(.*)"
```

Comment:
Would it be useful to match multiple whitespace characters after "optional, "in" or "out"?  Something like `(?:(?:optional|in|out) *)`